docs/adr/0006-process-for-onboarding-new-tenants.md

```diff
@@ -0,0 +1,270 @@
+ |
+ +-> WORKFLOW
+ |     |
+ |     +-> WORKFLOW+Litware
+ |           +-> (Operations storage configuration)
+ |           +-> (The Id of the OPERATIONS+WORKFLOW+Litware sub-tenant for the Operations service to use)
+ |
+ +-> OPERATIONS
+ |     |
+ |     +-> OPERATIONS+WORKFLOW+Litware
+ |
+ +-> FOOBAR
+```
+
+Now, the Operations service calls FooBar's enrollment endpoint to enroll the OPERATIONS+WORKFLOW+Litware tenant, again passing through the configuration it received when the Workflow service invoked it. The FooBar service has no dependencies so does not need to create any further tenants; it simply attaches it's storage configuration to the tenant being enrolled and returns. This also completes the WORKFLOW+Litware tenant's enrollment for Operations.
+
+```
+Root tenant
+ |
+ +-> Contoso
+ |
+ +-> Litware
+ |     +-> (Workflow storage configuration)
+ |     +-> (The Id of the WORKFLOW+Litware sub-tenant for the Workflow service to use)
+ |
+ +-> WORKFLOW
+ |     |
+ |     +-> WORKFLOW+Litware
+ |           +-> (Operations storage configuration)
+ |           +-> (The Id of the OPERATIONS+WORKFLOW+Litware sub-tenant for the Operations service to use)
+ |
+ +-> OPERATIONS
+ |     |
+ |     +-> OPERATIONS+WORKFLOW+Litware
+ |           +-> (FooBar storage configuration)
+ |
+ +-> FOOBAR
+```
+
+Next, the Workflow service continues it's enrollment process by enrolling the new WORKFLOW+Litware tenant in it's other dependency, FooBar. As with the Operations service enrolling OPERATIONS+WORKFLOW+Litware with FooBar, this does not result in any further tenants being created, just the FooBar config being attached to WORKFLOW+Litware:
+
+```
+Root tenant
+ |
+ +-> Contoso
+ |
+ +-> Litware
+ |     +-> (Workflow storage configuration)
+ |     +-> (The Id of the WORKFLOW+Litware sub-tenant for the Workflow service to use)
+ |
+ +-> WORKFLOW
+ |     |
+ |     +-> WORKFLOW+Litware
+ |           +-> (Operations storage configuration)
+ |           +-> (The Id of the OPERATIONS+WORKFLOW+Litware sub-tenant for the Operations service to use)
+ |           +-> (FooBar storage configuration)
+ |
+ +-> OPERATIONS
+ |     |
+ |     +-> OPERATIONS+WORKFLOW+Litware
+ |           +-> (FooBar storage configuration)
+ |
+ +-> FOOBAR
+```
+
+This completes Litware's enrollment for the Workflow service. As can be seen, this has resulted in multiple service-specific Litware tenants being created. However, there is a further step: Litware also needs to be enrolled in the Operations service. At present, it is able to indirectly use the service via Workflow, but also needs to be able to use it directly. So, the process we went through for Workflow is repeated: the management API requests the description of the configuration that must be provided to enroll Litware in the Operations service - which is storage configuration for Operations and FooBar - and then supplies that configuration to the Operations service's enrollment endpoint.
```

Comment:
See comment on line 260. I am concerned to make clear that the Client cannot *make use of* those services (directly or indirectly) but that those services are used on their behalf, and without their knowledge (even if they can infer it because of leaky implementation details such as not putting a facade over the operations API).

Reply:
I've added some additional wordage which hopefully reinforces that point.